### PR TITLE
libnifalcon: init at 1.1

### PR DESCRIPTION
**Pull request by wesleysliao**
###### Motivation for this change
A library for development using the Novint Falcon haptic device. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        load firmware to the internal microcontroller. In addition, it comes
+        with sample functionality made available through the firmware available
+        in NovInt’s drivers (the novint.bin file in TestUtilties and the
+        nifalcon_test_fw files for the library source). This firmware is
+        distributed in the firmware directory of the source distribution, and
+        is required for the findfalcons utility to run.
+      '';
+
+    maintainers = with maintainers; [ wesliao ];
+  };
```
Comment (ymarkus) [inline pkgs/development/libraries/libnifalcon/default.nix:35]: Please set meta.platforms, see [nixpkgs manual](https://nixos.org/nixpkgs/manual/#reviewing-contributions-new-packages) for details:
> Platforms should be set or the package will not get binary substitutes.

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  name = "libnifalcon-1.1";
+
+  src = fetchFromGitHub {
+    owner = "libnifalcon";
+    repo = "libnifalcon";
+    rev = "c1546ec8d9a94b3be7d74f8ba9f1a121a0ebb0c2";
+    sha256 = "1gp2cf56rlnlw785vl90w6scpknajw2mdny26yqlx5yl277x9b9s";
+  };
+
+  buildInputs = [ cmake libusb1 doxygen pkgconfig expat ];
```
Comment (ymarkus) [inline pkgs/development/libraries/libnifalcon/default.nix:13]: Are these all needed as run-time dependencies?
Build-time dependencies should be declared as `nativeBuildInputs`

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  outputs = [ "out" "dev" ];
+
+  meta = with stdenv.lib; {
+    homepage = "http://qdot.github.com/libnifalcon/";
+    description = "Open source drivers for the Novint Falcon haptic controller";
+    license = licenses.bsd3;
+
+    longDescription =
+      ''libnifalcon is a development library for the NovInt Falcon,
```
Comment (ymarkus) [inline pkgs/development/libraries/libnifalcon/default.nix:22]: Please move the `''` up behind longDescription, so it looks like this:
```
longDescription = ''
  libnifalcon is a development library for the NovInt Falcon,
  [...]
'';
```

**Review by ymarkus — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] phases are respected

##### Possible improvements
Please see my comments

##### Comments
Builds fine & all binaries can be run, although I can't test them more, as I don't have this device.

File: pkgs/development/libraries/libnifalcon/default.nix (hunk 1 above)
Comment (ymarkus) [reply pkgs/development/libraries/libnifalcon/default.nix:35]: `platforms = platforms.all;` should be fine, as the project states so itself.

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "libnifalcon";
+    rev = "c1546ec8d9a94b3be7d74f8ba9f1a121a0ebb0c2";
+    sha256 = "1gp2cf56rlnlw785vl90w6scpknajw2mdny26yqlx5yl277x9b9s";
+  };
+
+  buildInputs = [ cmake libusb1 doxygen pkgconfig expat ];
+  outputs = [ "out" "dev" ];
+
+  meta = with stdenv.lib; {
+    homepage = "http://qdot.github.com/libnifalcon/";
```
Comment (jtojnar) [inline pkgs/development/libraries/libnifalcon/default.nix:17]: Link does not seem to work.

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, cmake, doxygen, libusb1, pkgconfig, expat }:
+
+stdenv.mkDerivation rec {
+  name = "libnifalcon-1.1";
```
Comment (jtojnar) [inline pkgs/development/libraries/libnifalcon/default.nix:4]:
```suggestion
  pname = "libnifalcon";
  version = "1.1";
```

File: pkgs/development/libraries/libnifalcon/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "libnifalcon";
+    repo = "libnifalcon";
+    rev = "c1546ec8d9a94b3be7d74f8ba9f1a121a0ebb0c2";
```
Comment (jtojnar) [inline pkgs/development/libraries/libnifalcon/default.nix:9]:
```suggestion
    rev = version;
```

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -8779,6 +8779,12 @@
     email = "kirill.wedens@gmail.com";
     name = "wedens";
   };
+  wesliao = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8782]: Please put this into a separate commit with the message ``maintainers: add wesliao``.

File: pkgs/development/libraries/libnifalcon/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libnifalcon/default.nix:13]: at least pkg-config needs to move.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Closing due to unresponsive author. Feel free to reopen the discussion.
